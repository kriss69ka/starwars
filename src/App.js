import React from "react";
import "./App.css";

import InputSearch from "./InputSearch";
import {findHighest} from "./findHighest";

import data from "./Data.json";

class App extends React.Component {
  state = {
    filter: ""
  };

  handleChange = e => this.setState({ filter: e.target.value });

  filterdArr = arr =>
    arr.filter(item =>
      item.name.toLowerCase().includes(this.state.filter.toLowerCase())
    );

  render() {
    const maxHeight = findHighest(this.filterdArr(data));
    return (
      <div className="App">
        <InputSearch handleChange={this.handleChange} />
        <table>
          <tbody>
            <tr>
              <td>Name</td>
              <td>Hair color</td>
              <td>Height</td>
            </tr>
            {this.filterdArr(data).map(item => (
              <tr
                key={item.name}
                className={
                  (parseInt(item.height, 10) === maxHeight && "highest") || ""
                }
              >
                <td>{item.name}</td>
                <td>{item.hair_color}</td>
                <td>{item.height}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
export default App;
