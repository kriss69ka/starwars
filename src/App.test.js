import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { shallow } from "enzyme";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders", () => {
  expect(shallow(<App />).debug()).toMatchSnapshot();
  expect(
    shallow(<App />)
      .find(".highest")
      .find("td")
      .at(0)
      .text()
  ).toBe("Darth Vader");
});
