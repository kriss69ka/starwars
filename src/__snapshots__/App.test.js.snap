// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
"<div className=\\"App\\">
  <InputSearch handleChange={[Function]} />
  <table>
    <tbody>
      <tr>
        <td>
          Name
        </td>
        <td>
          Hair color
        </td>
        <td>
          Height
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          Luke Skywalker
        </td>
        <td>
          blond
        </td>
        <td>
          172
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          C-3PO
        </td>
        <td>
          n/a
        </td>
        <td>
          167
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          R2-D2
        </td>
        <td>
          n/a
        </td>
        <td>
          96
        </td>
      </tr>
      <tr className=\\"highest\\">
        <td>
          Darth Vader
        </td>
        <td>
          none
        </td>
        <td>
          202
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          Leia Organa
        </td>
        <td>
          brown
        </td>
        <td>
          150
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          Owen Lars
        </td>
        <td>
          brown, grey
        </td>
        <td>
          178
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          Beru Whitesun lars
        </td>
        <td>
          brown
        </td>
        <td>
          165
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          R5-D4
        </td>
        <td>
          n/a
        </td>
        <td>
          97
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          Biggs Darklighter
        </td>
        <td>
          black
        </td>
        <td>
          183
        </td>
      </tr>
      <tr className=\\"\\">
        <td>
          Obi-Wan Kenobi
        </td>
        <td>
          auburn, white
        </td>
        <td>
          182
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;
